### PR TITLE
Add a private API banner

### PR DESCRIPTION
**Pull request by svekars**
Add a banner that will appear on all pages where the last segment of the URL starts with an underscore "_". 
Example pages: 
* https://pytorch.org/docs/master/_dynamo.html
* https://pytorch.org/docs/master/_modules/torch/_jit_internal.html
Sample screenshots:
Screenshot 2023-02-03 at 1 13 47 PM
Screenshot 2023-02-03 at 1 12 51 PM

cc @carljparker

### CODE REVIEW

File: docs/source/_templates/layout.html [hunk 1]
```diff
@@ -31,6 +31,23 @@
...
+
+//const re = new RegExp("/_[a-zA-Z0-9_]*.html")
+//const match = window.location.href.match(re)
+
+var match      = window.location.href.match(/\/_[a-zA-Z0-9_]*.html|_dynamo/gi);
+var url  = window.location.href.lastIndexOf(match[match.length-1]);
+
+if (url)
+  {
+    var div = '<div class="admonition note"><p class="admonition-title">Note</p><p><i class="fa fa-exclamation-circle" aria-hidden="true">&nbsp</i> This page describes a private API. No support is provided for private APIs.</p></div>'
```
Comment (malfet) [inline docs/source/_templates/layout.html:46]: How about something along the following lines.
```suggestion
    var div = '<div class="admonition note"><p class="admonition-title">Note</p><p><i class="fa fa-exclamation-circle" aria-hidden="true">&nbsp</i> This is a private API, which is not supposed to be used outside of PyTorch codebase, and can be changed or removed without any notice.</p></div>'
```

**Review by malfet — APPROVED** [review]
Looks good to me, though the message is wrong imo

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**